### ISSUE TITLE
Edge nodes introduced even when preservation is requested

### ISSUE THREAD

**Issue by austinherrema** [dengwirda/mesh2d#10]
It seems that when a region has a very high aspect ratio (e.g. a triangle with a very small angle), nodes are inserted into the border even when I have `ref1` set to `preserve`. Is this the expected behavior? Is there anything I can do to be more strict about preserving edge nodes?

**Comment by dengwirda** [comment]
@austinherrema Yes, there are presently some cases where the boundary will be split regardless -- as you say, this should generally only happen adjacent to sharp features. 

Refinement-style algorithms (like those in `MESH2D`) need to ensure that sharp boundary features are "protected" in some way (otherwise the refinement won't converge). This generally involves forming isosceles triangles in the apex of any sharp angles in the geometry. I suspect this is where the boundary splitting you mention is coming from.

It may be possible to do something different in these cases when the `preserve` option is selected, but I'll need to think a bit more thoroughly on this.

If it's possible to send an example geometry this would be a help.

**Comment by austinherrema** [comment]
Makes sense. I put together an [example geomety/script](https://github.com/dengwirda/mesh2d/files/2931956/example.zip) that demonstrates this behavior. Let me know if you would like further info!

![image](https://user-images.githubusercontent.com/29387274/53823244-99081500-3f2e-11e9-9f08-e03eb5e59fa8.png)

**Comment by dengwirda** [comment]
@austinherrema Thanks for the example -- I think this is definitely due to the "protection" business I described.

I'll see whether anything different can be done in these type of cases, but this may require a bit of a re-think of the algorithm's logic. 

In this particular case though, if absolutely no new nodes were inserted anywhere on the boundary, you would end up with some (extremely!) poor quality triangles -- is this really the desired output?

**Comment by austinherrema** [comment]
@dengwirda Sorry for the late response. I do understand that the end result would be extremely poor quality triangles. However, in my case the edge nodes are used to establish connectivity to bordering polygons which are meshed independently (I do know mesh2d has capabilities for meshing multiple regions simultaneously--I had some other issues with that which I guess would be another topic). When border nodes are inserted, and because I cannot easily track where they are inserted, it is difficult for me to maintain connectivity to bordering polygons. This lack of connectivity is more problematic for me than the poor quality triangles, which I can clean up using my own methods later on.

I understand if this is simply a limitation of the approach in mesh2d right now.
